src/components/QueryForm.tsx

```diff
@@ -1,179 +1,97 @@
-import { useState, useEffect } from 'react';
-import { useSearchParams } from 'react-router-dom';
-import { Button, FormControlLabel, Checkbox } from '@mui/material';
+import { useState } from 'react';
+import {
+  Button,
+  FormControlLabel,
+  Checkbox,
+  CircularProgress,
+  FormHelperText,
+} from '@mui/material';
 import SendIcon from '@mui/icons-material/Send';
 import { isFederationAPI, sexes, modalities } from '../utils/constants';
-import { FieldInput, FieldInputOption, NodeOption, AttributeOption } from '../utils/types';
+import { NodeOption, AttributeOption, FieldInput } from '../utils/types';
 import CategoricalField from './CategoricalField';
 import ContinuousField from './ContinuousField';
 
 function QueryForm({
   nodeOptions,
   diagnosisOptions,
   assessmentOptions,
-  apiQueryURL,
+  node,
+  minAge,
+  maxAge,
+  sex,
+  diagnosis,
+  isControl,
+  setIsControl,
+  assessmentTool,
+  imagingModality,
+  updateCategoricalQueryParams,
+  updateContinuousQueryParams,
+  loading,
   onSubmitQuery,
 }: {
   nodeOptions: NodeOption[];
   diagnosisOptions: AttributeOption[];
   assessmentOptions: AttributeOption[];
-  apiQueryURL: string;
-  onSubmitQuery: (url: string) => void;
+  node: FieldInput;
+  minAge: number | null;
+  maxAge: number | null;
+  sex: FieldInput;
+  diagnosis: FieldInput;
+  isControl: boolean;
+  setIsControl: (value: boolean) => void;
+  assessmentTool: FieldInput;
+  imagingModality: FieldInput;
+  updateCategoricalQueryParams: (label: string, value: FieldInput) => void;
+  updateContinuousQueryParams: (label: string, value: number | null) => void;
+  loading: boolean;
+  onSubmitQuery: () => void;
 }) {
-  const [node, setNode] = useState<FieldInput>([{ label: 'All', id: 'allNodes' }]);
-  const [minAge, setMinAge] = useState<string | null>(null);
-  const [maxAge, setMaxAge] = useState<string | null>(null);
-  const [sex, setSex] = useState<FieldInput>(null);
-  const [diagnosis, setDiagnosis] = useState<FieldInput>(null);
-  const [isControl, setIsControl] = useState<boolean>(false);
-  const [minNumSessions, setMinNumSessions] = useState<string | null>(null);
-  const [assessmentTool, setAssessmentTool] = useState<FieldInput>(null);
-  const [imagingModality, setImagingModality] = useState<FieldInput>(null);
-  const [searchParams, setSearchParams] = useSearchParams();
+  const [minAgeError, setMinAgeError] = useState(false);
+  const [minAgeHelperText, setMinAgeHelperText] = useState('');
+  const [maxAgeError, setMaxAgeError] = useState(false);
+  const [maxAgeHelperText, setMaxAgeHelperText] = useState('');
+  const [minNumSessionsError, setMinNumSessionsError] = useState(false);
+  const [minNumSessionsHelperText, setMinNumSessionsHelperText] = useState('');
```

Comment:
I'm not sure why any of these are stateful. All of these can be computed directly from the props - and the props are handled by the parent now (i.e. App.tsx). So unless I'm missing something, the `validate` function can go, and the only eventhandler here is the forwarded `updateCategoricalOption` or `updateContinuousOption` handlers from the parent